### PR TITLE
Fix init io handling

### PR DESCRIPTION
**Pull request by garazdawi**
This fixes a bug introduced in 95dcdb7754.

While at it, now that I better understand what is going on, I cleaned up the code in init a bit and added a short-circuit in application_master to make io handling a little bit more performant.

### CODE REVIEW

File: lib/kernel/src/application_master.erl [hunk 1]
```diff
@@ -162,7 +162,7 @@ start_it(State, Type) ->
 init_loop(Pid, Tag, State, Type) ->
     receive
  	IoReq when element(1, IoReq) =:= io_request ->
-            relay_to_group_leader(IoReq, State),
+            _ = relay_to_group_leader(IoReq, State),
```
Comment (ranjanified) [inline lib/kernel/src/application_master.erl:165]: Any particular reason for this '_' binding? With the notion of  "function return value(s)" I can understand, anything else apart from that? Only curious.

Comment (garazdawi) [reply lib/kernel/src/application_master.erl:165]: It is used to suppress a dialyzer warning regarding unmatched returns.

Comment (ranjanified) [reply lib/kernel/src/application_master.erl:165]: O!! my my dear little brain... Thanks. :smiley:

**Comment by garazdawi** [comment]
merged to master

**Pull request closed without merging**